### DISCUSSION TITLE
Why destination chain not execute proposal once threshold met?

### DISCUSSION THREAD

**Discussion by livelybug** [ChainSafe/chainbridge-core#319] [General]
Hello,

I am reading the doc Transfer Flow --> As Destination Chain(https://chainbridge.chainsafe.io/chains/ethereum/#as-destination-chain).

I have a question about the flow.
Once a destination chain has met some vote threshold for a proposal, why doesn’t the chain execute the proposal immediately?
It would save the relayers' gas fee of calling executeDeposit.

Thank you

**Comment by roethke** [comment]
Hi @livelybug, thanks for your question. Are you suggesting that the contract itself perform the action to execute the proposal rather than the relayer?

**Reply by livelybug** [reply]
> Hi @livelybug, thanks for your question. Are you suggesting that the contract itself perform the action to execute the proposal rather than the relayer?

Hi, Thanks for your response. Yes, that's exactly my suggestion. Is the suggestion viable for the chainbridge project? Is there any security concern about it? Thanks

**Reply by roethke** [reply]
Unfortunately contracts need to be _poked_ in order for actions to be taken. Because of this limitation, the relayer manages the invocation of contract methods such as executing a proposal to mint tokens on the destination chain.

**Reply by livelybug** [reply]
Notice the latest [chainbridge-solidity](https://github.com/ChainSafe/chainbridge-solidity) already adopted MPC and removed voting logic.

My question is still about the voting logic and may be outdated 

Tracing back to [the commit right before the voting logic](https://github.com/ChainSafe/chainbridge-solidity/blob/6bcfdb19aa) is removed, [the proposal is executed once the threshold is met](https://github.com/ChainSafe/chainbridge-solidity/blob/6bcfdb19aaac0bc62bf3134f3fc26c854265e401/contracts/Bridge.sol#L434). which is upgraded from [commit develop](https://github.com/ChainSafe/chainbridge-solidity/commit/8979e9e242add51b97e2eebaa221cf513e5bdde6#diff-d641d27f43a4e1d3dc731dff26fbbaf62e42388ae1455ec70b9521a1e6f27868L363-R372), where ```voteProposal``` and ```executeProposal``` were merged to be called by one tx instead of two. 
So the doc(https://chainbridge.chainsafe.io/chains/ethereum/#as-destination-chain) is outdated by the code. The relayer only needs one tx in order for a proposal to be executed.
Feel free to correct me if wrong. Thanks

**Reply by roethke** [reply]
Chainbridge-solidity does not include MPC code; as of now, MPC is specific only to Sygma but yes, the chainbridge docs need to be updated in general.

In general though, the bridge contract would need to be watching for the event or state change in order to then execute the proposal, which cannot be managed without a third party software like a relayer watching for events and submitting txs to the bridge contract, though perhaps you knew as much already.

I hope this answers your question. If you have a Sygma-related question, feel free to open up an issue within either the [sygma-relayer](https://github.com/sygmaprotocol/sygma-relayer) or the [sygma-solidity](https://github.com/sygmaprotocol/sygma-solidity) repositories. We will be activating discussions functionality on those repositories shortly.

**Reply by livelybug** [reply]
Thank you, that does answer my question.
